Applications/Terminal/Terminal.cpp

```diff
@@ -837,52 +972,37 @@ void Terminal::paint_event(GPaintEvent& event)
 
     GPainter painter(*this);
 
-    if (m_needs_background_fill) {
-        m_needs_background_fill = false;
-        if (m_visual_beep_timer.is_active())
-            painter.fill_rect(frame_inner_rect(), Color::Red);
-        else
-            painter.fill_rect(frame_inner_rect(), Color(Color::Black).with_alpha(255 * m_opacity));
-    }
-
-    if (m_rows_to_scroll_backing_store && m_rows_to_scroll_backing_store < m_rows) {
```

Comment:
Did you establish whether or not there was actually any problem with the repaint optimization after adding the missing escape handling? It would be a shame to lose it unnecessarily.